src/Avalonia.Controls/Primitives/Popup.cs

```diff
@@ -128,9 +128,22 @@ public class Popup : Control, IVisualTreeHost
         public static readonly StyledProperty<bool> TopmostProperty =
             AvaloniaProperty.Register<Popup, bool>(nameof(Topmost));
 
+        /// <summary>
+        /// Defines the <see cref="ShouldConstrainToRootBounds"/> property.
+        /// </summary>
+        public static readonly StyledProperty<bool> ShouldConstrainToRootBoundsProperty =
```

Comment:
Personally I don't like the naming. Something like "ShouldUsePopupHost" or "UseOverlayLayer" will be more accurate.
Moreover, related issue https://github.com/AvaloniaUI/Avalonia/issues/5628 makes even less sense in "ConstrainToRootBounds" name.

Reply:
But I guess we prefer UWP/WPF API anyways.

Reply:
I always like using the 'Is' prefix myself but there are exceptions where it just doesn't make grammatical sense. Also remember those working on WinUI now don't really follow the original naming conventions as much as they should.

That said, 'RootBounds' makes more sense than PopupHost or OverlayLayer. The name shouldn't imply how this functionality is implemented, it should imply what the functionality does in an abstract way.

'IsConstrainToRootBoundsRequested' or 'IsConstrainToRootBoundsEnabled' might be more conventional, but verbose, names.

Reply:
Okay, I agree about hiding implementation details in that case.